### PR TITLE
TS migrations

### PR DESCRIPTION
**Pull request by ebarakos**
Migrating the following adapters to Typescript:

`google-finance`
`kaiko`
`marketstack`
`metalsapi`
`nikkei`
`nomics`
`openexchangerates`
`poa-gasprice`
`polygon`

### CODE REVIEW

File: kaiko/src/index.ts
Comment (justinkaseman) [inline kaiko/src/index.ts:5]:
```suggestion
const NAME = 'KAIKO'
```

@krebernisak Do we want this capitalized or lowercase? I've seen both ways, but let's pick a convention.

File: kaiko/src/config.ts
Comment (justinkaseman) [inline kaiko/src/config.ts:4]: This is currently an unused variable as there is only one endpoint.

I would instead make every EA similar to the `example` adapter's  `adapter.ts` code with the switch statement routing the logic, even if there is only one endpoint. It makes it a lot easier to extend in the future.

File: marketstack/package.json
Comment (justinkaseman) [inline marketstack/package.json:31]:
```suggestion
```
Take these out of every TS migrated EA as `./index.js` is now gone.

File: marketstack/src/adapter.ts
Comment (justinkaseman) [inline marketstack/src/adapter.ts:23]: This will quickly grow to be hard to read if say 10 other endpoints were added.

I think the best style would be how Jonas modified the `example` adapter as seen here: 
https://github.com/smartcontractkit/external-adapters-js/blob/develop/example/src/endpoint/example.ts#L32
Each endpoint handles its own return, which keeps `adapter.ts` a lot cleaner.

File: marketstack/src/endpoint/eod.ts
Comment (justinkaseman) [inline marketstack/src/endpoint/eod.ts:8]: @krebernisak Should we be adding typing for each API response shape? Seems like it might be hard to maintain, but could provide valuable debugging information.

File: marketstack/src/endpoint/eod.ts
Comment (justinkaseman) [inline marketstack/src/endpoint/eod.ts:35]: I would expect the `baseURL` and `API_KEY` to be the same no matter what endpoint is being hit. These might be better injected in Config.ts,

File: marketstack/src/index.ts
Comment (justinkaseman) [inline marketstack/src/index.ts:5]:
```suggestion
const NAME = 'MARKETSTACK'
```

File: metalsapi/package.json
Comment (justinkaseman) [inline metalsapi/package.json:3]: Should we be bumping versions for all of these TS migrations?

File: metalsapi/src/index.ts
Comment (justinkaseman) [inline metalsapi/src/index.ts:5]:
```suggestion
const NAME = 'METALSAPI'
```

File: nikkei/src/index.ts
Comment (justinkaseman) [inline nikkei/src/index.ts:5]:
```suggestion
const NAME = 'NIKKEI'
```

File: nomics/src/index.ts
Comment (justinkaseman) [inline nomics/src/index.ts:5]:
```suggestion
const NAME = 'NOMICS'
```

File: openexchangerates/src/config.ts
Comment (justinkaseman) [inline openexchangerates/src/config.ts:4]: As with the other EA I think this should be changed to be used as other `endpoint`s are.

**Review by justinkaseman — CHANGES_REQUESTED** [review]
Great!

Added a few small tweaks.

One thing I am unsure of is what the `endpoint` would be for some of these adapters. I know we want to keep similar endpoints conformed to the same name and params. Just thinking aloud that we need them documented somewhere.

File: kaiko/src/index.ts
Comment (krebernisak) [reply kaiko/src/index.ts:5]: This was first used as an ENV prefix to avoid naming collisions when using two or more adapters in the same env. So for now it was a simple hardcoded uppercase string

What we should do now to make this more robust is this:

- Import the `name` from `package.json`, remove the `-adapter` suffix, and export that as NAME.
- Create a `toEnvName` util function that will map it to a safe string to use as ENV name

> Environment variable names used by the utilities in the Shell and Utilities volume of IEEE Std 1003.1-2001 consist solely of uppercase letters, digits, and the '_' (underscore) from the characters defined in Portable Character Set and do not begin with a digit.

File: metalsapi/package.json
Comment (krebernisak) [reply metalsapi/package.json:3]: Yeah, I think we should!

File: marketstack/src/endpoint/eod.ts
Comment (krebernisak) [reply marketstack/src/endpoint/eod.ts:8]: Not in this PR to keep it fast & simple.

But in the future, we should type anything that's useful, and I think that will become useful at one point. The further we go with composite adapters, the more we use the data provider adapters programmatically, the more useful we are going to find those types.

First, we should finish this migration, then focus on improvements to our internal framework data types, and then we can make another push like this where we add a bunch of underlying API data types in one push. At that point it could start being useful :)

File: marketstack/package.json
Comment (ebarakos) [reply marketstack/package.json:31]: Taking it out of the `example` as well.

File: metalsapi/package.json
Comment (ebarakos) [reply metalsapi/package.json:3]: Since all versions are now set to `0.0.1`, let's wait until everything is finished/merged, and batch-update all TS migrated ones, or all TS ones in general.

**Comment by ebarakos** [comment]
Converted this to draft as it had been split to 3 smaller batches. To be closed after merging the migrations.

**Pull request closed without merging**
